plugins/outputs/nats/nats_test.go

```diff
@@ -263,3 +266,230 @@ func createStream(t *testing.T, server string, cfg *nats.StreamConfig) {
+			},
+			fields: map[string]interface{}{
+				"cpu_usage":  1,
+				"cpu_idle":   2,
+				"cpu_system": 3,
+			},
+			msgCount: 4, // Our 3 fields plus the default value field
+			expectedSubjects: []string{
+				"my-subject.metrics.foo.bar.test1.cpu_usage",
+				"my-subject.metrics.foo.bar.test1.cpu_idle",
+				"my-subject.metrics.foo.bar.test1.cpu_system",
+				"my-subject.metrics.foo.bar.test1.value",
+			},
+		},
+		{
+			name: "subject layout missing end tag",
+			container: testutil.Container{
+				Image:        "nats:latest",
+				ExposedPorts: []string{natsServicePort},
+				Cmd:          []string{"--js"},
+				WaitingFor:   wait.ForListeningPort(nat.Port(natsServicePort)),
+			},
+			nats: &NATS{
+				Name:    "telegraf",
+				Subject: "my-subject.{{ .Name }}.{{ .Tag \"tag1\" }}.{{ .Tag \"tag2\" }}",
+				Jetstream: &StreamConfig{
+					Name:     "my-telegraf-stream",
+					Subjects: []string{"my-subject.>"},
+				},
+				serializer: &influx.Serializer{},
+				Log:        testutil.Logger{},
+			},
+			streamConfigCompareFunc: func(t *testing.T, si *jetstream.StreamInfo) {
+				require.Equal(t, "my-telegraf-stream", si.Config.Name)
+				require.Equal(t, []string{"my-subject.>"}, si.Config.Subjects)
+			},
+			tags: map[string]string{
+				"tag1": "foo",
+			},
+			msgCount: 0,
+		},
+		{
+			name: "subject layout missing missing subjects",
+			container: testutil.Container{
+				Image:        "nats:latest",
+				ExposedPorts: []string{natsServicePort},
+				Cmd:          []string{"--js"},
+				WaitingFor:   wait.ForListeningPort(nat.Port(natsServicePort)),
+			},
+			nats: &NATS{
+				Name:    "telegraf",
+				Subject: "my-subject.{{ .Name }}.{{ .Tag \"tag1\" }}.{{ .Tag \"tag2\" }}",
+				Jetstream: &StreamConfig{
+					Name: "my-telegraf-stream",
+				},
+				serializer: &influx.Serializer{},
+				Log:        testutil.Logger{},
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.container.Start()
+			require.NoError(t, err, "failed to start container")
+			defer tc.container.Terminate()
+
+			server := []string{fmt.Sprintf("nats://%s:%s", tc.container.Address, tc.container.Ports[natsServicePort])}
+			tc.nats.Servers = server
+			// Verify that we can connect to the NATS daemon
+			err = tc.nats.Init()
+			if tc.wantErr {
+				require.Error(t, err)
+				return
+			}
+			err = tc.nats.Connect()
+			require.NoError(t, err)
+
+			if tc.nats.Jetstream != nil {
+				stream, err := tc.nats.jetstreamClient.Stream(t.Context(), tc.nats.Jetstream.Name)
+				require.NoError(t, err)
+				si, err := stream.Info(t.Context())
+				require.NoError(t, err)
+
+				tc.streamConfigCompareFunc(t, si)
+			}
+			// Verify that we can successfully write data to the NATS daemon
+			metric := testutil.MockMetrics()
+			for _, m := range metric {
+				for k, v := range tc.tags {
+					m.AddTag(k, v)
+				}
+				for k, v := range tc.fields {
+					m.AddField(k, v)
+				}
+			}
```

Comment:
Please just define a set of metrics suited to your test case! It is really hard to read this test if you need to scroll up and down to find out what the input data is an what you expect...

Reply:
Fixed.